File: 백준/Silver/10828. 스택/스택.py
import sys

class Stack:
    def __init__(self):
        self.items = []

    def push(self, val):
        self.items.append(val)

    def pop(self):
        try:
            print(self.items[-1])
            self.items.pop()
        except IndexError:
            print(-1)

    def top(self):
        try:
            print(self.items[-1])
        except IndexError:
            print(-1)

    def empty(self):
        if len(self.items) == 0:
            print(1)
        else:
            print(0)

    def size(self):
        print(len(self.items))

s = Stack()

n = int(sys.stdin.readline())
stackList = [sys.stdin.readline().strip() for i in range(n)]


for i in range(n):
    if stackList[i][1] == 'u': # push
        s.push(stackList[i][5:])
    elif stackList[i][0] == 'p' and stackList[i][1] == 'o': # pop
        s.pop()
    elif stackList[i][0] == 't': # top
        s.top()
    elif stackList[i][0] == 's': # size
        s.size()
    elif stackList[i][0] == 'e': # empty
        s.empty()

File: 백준/Silver/10828. 스택/스택.js
let fs = require('fs');
let input = fs.readFileSync(process.platform === "linux" ? "/dev/stdin" : "./input.txt")
    .toString()
    .trim()
    .split('\n')

const stack = [];

for (let i=1; i<=input[0]; i++) {
    const temp = input[i].split(' ');
    const cmd = temp[0];

    if (temp.length > 1) {
        const num = temp[1];
        stack.push(num);
    } else {
        if (cmd === 'pop') {
            if (stack.length === 0) {
                console.log(-1);
            } else {
                console.log(stack.pop());
            }
        } else if (cmd === 'size') {
            console.log(stack.length);
        } else if (cmd === 'empty') {
            if (stack.length === 0) {
                console.log(1);
            } else {
                console.log(0);
            }
        } else if (cmd === 'top') {
            if (stack.length === 0) {
                console.log(-1)
            } else {
                console.log(stack[stack.length-1]);
            }
        }
    }
}